src/Components/TodoList/TodoList.tsx

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import { Todo } from '../../types/Todo';
+import { TodoItem } from '../TodoItem';
+
+interface Props {
+  todos: Todo[],
+  onDelete: (id: number) => void;
+  newTodo: Todo | null;
```

Comment:
Not sure why you need `newTodo`...
After creating and adding a new todo item, you should add it to the main `todos` list to make sure that all new items appear in the list, not just the latest ones.
I think you have a problem now when you try to add multiple todo items in a row - please check this to understand the problem live and fix it